### PR TITLE
Migrate to GitHub actions from Travis.

### PR DESCRIPTION
**Pull request by FireMasterK**
- [x] I carefully read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md) and agree to them.
- [x] I have tested the API against [NewPipe](https://github.com/TeamNewPipe/NewPipe).
- [x] I agree to create a pull request for [NewPipe](https://github.com/TeamNewPipe/NewPipe) as soon as possible to make it compatible with the changed API.

I have not tested this yet, but everything should be fine I think.

@TheAssassin

### CODE REVIEW

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+name: Build and deploy JavaDocs
+
+on:
+  push:
+    branches:
+      - dev
```
Comment (TobiGr) [inline .github/workflows/docs.yml:6]: Docs should only change when releasing a new version
```suggestion
      - master
```

**Review by TobiGr — COMMENTED** [review]
Thank you. Code looks good to me but I did not test it yet.

File: .github/workflows/docs.yml (hunk 1 above)
Comment (FireMasterK) [reply .github/workflows/docs.yml]: Changed 👍

**Comment by TobiGr** [comment]
Please change the badge in README.md
See TeamNewPipe/NewPipe#5278

**Comment by FireMasterK** [comment]
Updated the README badge.

**Comment by TobiGr** [comment]
I used that code in my fork to test it. See https://github.com/TobiGr/NewPipeExtractor/runs/1630857048

As you can see, the workflow is cancelled as soon as the test job fails. That is bad, because there are always failing tests.

**Comment by TobiGr** [comment]
@FireMasterK Can you give us access to your branch so we can apply some fixes and merge afterwards? (E.g. some JDoc needs to be fixed before deployment works).
You can give us access to that branch by ticking a checkbox in the right sidebar.

**Comment by FireMasterK** [comment]
I don't have that option, likely due to my fork being in an org, I'll close this PR in favor of #506 which has those fixes 😃

**Pull request closed without merging**